src/components/dashboard/Claim.js

```diff
@@ -127,54 +107,30 @@ const Claim = props => {
     if (process.env.NODE_ENV !== 'test') {
       setLoading(true)
     }
-
-    await Promise.all([
-      goodWallet
-        .checkEntitlement()
-        .then(entitlement => setClaimState(prev => ({ ...prev, entitlement: entitlement.toNumber() })))
-        .catch(exception => {
-          const { message } = exception
-          const uiMessage = decorate(exception, ExceptionCode.E2)
-
-          log.error('gatherStats failed', message, exception, {
-            dialogShown: true,
-            category: ExceptionCategory.Blockhain,
-          })
-
-          showErrorDialog(uiMessage, '', {
-            onDismiss: () => {
-              screenProps.goToRoot()
-            },
-          })
-        }),
-      evaluateFRValidity(),
-    ])
+    await evaluateFRValidity()
     setLoading(false)
   }
 
   useEffect(() => {
     init()
+    setClaimInterval(setInterval(gatherStats, 1000))
```

Comment:
not sure this will works correctly. gatherStats may get old props / state values 
please look at this example of useInterval - they're using ref for save & update current callback

https://github.com/donavon/use-interval/blob/master/src/index.js

we could just add this hook to lib/hooks & reuse it

Reply:
gatherstats only sets props and its working ok
i'll fix the interval issue